cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIMongoBaseSinkTest.java

```diff
@@ -79,83 +82,194 @@ public void testConfigureCollectionPrefixIsNotSystem() {
+                        + "- FAIL - '" + collectionName + "' was crated as collection name instead of 'sth_/'");
+                throw e;
+            } // try catch
+        } catch (Exception e) {
+            System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
+                        + "- FAIL - There was some problem when building the collection name");
+            assertTrue(false);
+        } // catch
+    } // testBuildCollectionNameDMByServicePathRootServicePathOldEncoding
     
     /**
      * [NGSIMongoBaseSink.buildCollectionName] -------- When / service-path is notified/defaulted and
-     * data_model=dm-by-service-path, the MongoDB collection name is <prefix>/.
+     * data_model=dm-by-service-path, the MongoDB collection name is the concatenation of \<prefix\> and
+     * \<service-path\>.
      */
     @Test
-    public void testBuildCollectionNameDMByServicePathRootServicePath() {
+    public void testBuildCollectionNameDMByServicePathRootServicePathNewEncoding() {
         System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                 + "-------- When / service-path is notified/defaulted and data_model=dm-by-service-path, "
-                + "the MongoDB collection name is <prefix>/");
+                + "the MongoDB collection name is the concatenation of <prefix> and <service-path>");
         String collectionPrefix = "sth_";
         String dbPrefix = "sth_";
         String dataModel = "dm-by-service-path";
+        String enableEncoding = "true";
         NGSIMongoBaseSinkImpl sink = new NGSIMongoBaseSinkImpl();
-        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel));
+        sink.configure(CommonUtilsForTests.createContextForMongoSTH(collectionPrefix, dbPrefix, dataModel,
+                enableEncoding));
         String dbName = "sth_default";
         String fiwareService = "default";
         String fiwareServicePath = "/";
-        String entity = "someId_someType";
-        String attribute = "someName_someType";
+        String entity = "someId=someType";
+        String attribute = "someName";
         boolean isAggregated = false;
         String entityId = "someId";
         String entityType = "someType";
         
         try {
             String collectionName = sink.buildCollectionName(dbName, fiwareServicePath, entity, attribute,
                     isAggregated, entityId, entityType, fiwareService);
+            String expectedCollectionName = "sth_x002f";
             
             try {
-                assertEquals("sth_/", collectionName);
+                assertEquals(expectedCollectionName, collectionName);
                 System.out.println(getTestTraceHead("[NGSIMongoBaseSink.buildCollectionName]")
                         + "-  OK  - '" + collectionName + "' was crated as collection name");
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4